plugins/scenario/hu.bme.mit.gamma.scenario.model/src/hu/bme/mit/gamma/scenario/model/reduction/SimpleScenarioGenerator.java

```diff
@@ -143,8 +143,8 @@ private InitialBlock handleInitBlockCopy() {
 			return null;
 		}
 		InitialBlock initBloc = factory.createInitialBlock();
-		for (ModalInteraction modalInteraction : base.getInitialblock().getModalInteractions()) {
-			initBloc.getModalInteractions().add((ModalInteraction) doSwitch(modalInteraction));
+		for (InteractionDefinition interaction : base.getInitialblock().getInteractions()) {
+			initBloc.getInteractions().add((InteractionDefinition) doSwitch(interaction));
```

Comment:
Is the `(InteractionDefinition)` casting right in front of `doSwitch(interaction)` necessary?

Reply:
Yes, as the doSwitch function returns an EObject